### PR TITLE
sys-process/wait_on_pid: update EAPI 6 -> 8

### PR DESCRIPTION
**Pull request by Xpeq7**
Bug: https://bugs.gentoo.org/887019

### CODE REVIEW

**Comment by Xpeq7** [comment]
Bug: https://bugs.gentoo.org/887019

**Comment by N-R-K** [comment]
AFAIK, this shouldn't result in a rev-bump because it's fixing a build-time issue not a run-time one. From [the dev manual](https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html):

> Examples of changes that can be done without a revision bump are:
>
> * adding a patch to fix a build-time issue that prevented users from building the package (since it does not affect users who already managed to build it) unless: it affected runtime behaviour in some way

**Comment by Xpeq7** [comment]
OK, so, what should I do now?

**Comment by N-R-K** [comment]
> Is it enough, if I integrate the changes from (unwanted revision 2) into revision 1?

I think that should be fine. But I'm not a maintainer here, so you'll need to wait til a dev takes a look.

Also consider squashing all the commits into a single commit.

**Comment by Xpeq7** [comment]
done, didn't know how to do it from linux, so it took a bit longer.

**Comment by N-R-K** [comment]
> done, didn't know how to do it from linux, so it took a bit longer.

I typically do it with either `git rebase -i` or `git reset`.

File: sys-process/wait_on_pid/wait_on_pid-0.0.2-r1.ebuild [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="6"
+EAPI="8"
```
Comment (juippis) [inline sys-process/wait_on_pid/wait_on_pid-0.0.2-r1.ebuild:4]: Now this does require a revbump and dropping the keywords. Then I see the ebuild has more cleaning that we can do while updating it. 

But is this package still functional? Should it be kept around?

Comment (Xpeq7) [reply sys-process/wait_on_pid/wait_on_pid-0.0.2-r1.ebuild:4]: That's what I thought last time.
Is it functional? - I don't know, it's pulled in as a dependency for media-tv/gentoo-vdr-scripts-3.0_rc2-r1.
Should it be kept around - to be honest I don't know if it's that useful.
Does it work with EAPI="8"? Yes, just tested it.

I'm completely new to making these types of changes, just figuring out how to make this pull request took me 2 hours.
Thanks for letting me know, that this actually requires a revbump.

**Comment by Xpeq7** [comment]
tried my best to reverse the commit that changed -r1.
If I have to change anything else please let me know.

File: sys-process/wait_on_pid/wait_on_pid-0.0.2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+HOMEPAGE="https://dev.gentoo.org/~zzam/wait_on_pid/"
+SRC_URI="mirror://gentoo/${P}.tar.bz2 https://dev.gentoo.org/~zzam/${PN}/${P}.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~riscv ~x86 ~amd64-linux ~x86-linux"
+IUSE=""
+
+DEPEND=""
+RDEPEND=""
```
Comment (juippis) [inline sys-process/wait_on_pid/wait_on_pid-0.0.2-r2.ebuild:18]: These empty variables can be dropped to clean the ebuild a bit.

**Review by juippis — APPROVED** [review]
Not "quite", there's no reason to edit the -r1 ebuild. In other words, you maybe wanted to rebase and "squash" your 3 commits into one. But this is fine, I can either rebase or cherry-pick the latest commit and both will work. Thanks!

**Pull request closed without merging**